src/Layers/xrRenderPC_GL/r2_R_sun.cpp

```diff
@@ -214,8 +205,43 @@ Fvector3 wform(Fmatrix& m, Fvector3 const& v)
     r.w = v.x * m._14 + v.y * m._24 + v.z * m._34 + m._44;
     // VERIFY		(r.w>0.f);
     float invW = 1.0f / r.w;
-    Fvector3 r3 = {r.x * invW, r.y * invW, r.z * invW};
-    return r3;
+    return {r.x * invW, r.y * invW, r.z * invW};
+}
+
+Fvector3 wform(Fmatrix& m, glm::vec3 const& v)
+{
+    Fvector4 r;
+    r.x = v.x * m._11 + v.y * m._21 + v.z * m._31 + m._41;
+    r.y = v.x * m._12 + v.y * m._22 + v.z * m._32 + m._42;
+    r.z = v.x * m._13 + v.y * m._23 + v.z * m._33 + m._43;
+    r.w = v.x * m._14 + v.y * m._24 + v.z * m._34 + m._44;
+    // VERIFY		(r.w>0.f);
+    float invW = 1.0f / r.w;
+    return {r.x * invW, r.y * invW, r.z * invW};
+}
+
+Fvector3 wform(glm::mat4& m, Fvector3 const& v)
+{
+    Fvector4 r;
+    r.x = v.x * m[0][0] + v.y * m[1][0] + v.z * m[2][0] + m[3][0];
+    r.y = v.x * m[0][1] + v.y * m[1][1] + v.z * m[2][1] + m[3][1];
+    r.z = v.x * m[0][2] + v.y * m[1][2] + v.z * m[2][2] + m[3][2];
+    r.w = v.x * m[0][3] + v.y * m[1][3] + v.z * m[2][3] + m[3][3];
+    // VERIFY		(r.w>0.f);
+    float invW = 1.0f / r.w;
+    return {r.x * invW, r.y * invW, r.z * invW};
+}
+
+Fvector3 wform(glm::mat4& m, glm::vec3 const& v)
```

Comment:
I don't see that `glm::mat4& m` is changed anywhere in this function. It should be made const reference then.